File: benchmarks/Java/src/richards/Packet.java
/*
 * This benchmark is derived from Mario Wolczko's Java and Smalltalk version of
 * Richards.
 *
 * It is modified based on the SOM version and to use Java 8 features.
 * License details:
 *   http://web.archive.org/web/20050825101121/http://www.sunlabs.com/people/mario/java_benchmarking/index.html
 */
package richards;


final class Packet extends RBObject {
  public static final int DATA_SIZE = 4;

  Packet(final Packet link, final int identity, final int kind) {
    this.link     = link;
    this.identity = identity;
    this.kind     = kind;
    this.datum    = 0;
    this.data     = new int[DATA_SIZE];
  }

  private Packet      link;
  private int         identity;
  private final int   kind;
  private int         datum;
  private final int[] data;

  public int[] getData() { return data; }
  public int   getDatum() { return datum; }
  public void  setDatum(final int someData) { datum = someData; }

  public int  getIdentity() { return identity; }
  public void setIdentity(final int anIdentity) { identity = anIdentity; }

  public int getKind() { return kind; }
  public Packet getLink() { return link; }
  public void setLink(final Packet aLink) { link = aLink; }

  @Override
  public String toString() {
    return "Packet id: " + identity + " kind: " + kind;
  }
}

File: benchmarks/C++/src/richards/Packet.cpp
#include "Packet.h"
#include <utility>


namespace richards {
    Packet::Packet(shared_ptr<Packet> link, int identity, int kind) {
        _link = link;
        _identity = identity;
        _kind = kind;
        _datum = 0;
        _data = new int[DATA_SIZE];
    }

    int* Packet::getData() { 
        return _data; 
    }

    int Packet::getDatum() const {
        return _datum; 
    }

    void Packet::setDatum(int someData) { 
        _datum = someData; 
    }


    int Packet::getIdentity() const {
        return _identity; 
    }
            
    void Packet::setIdentity(int anIdentity) { 
        _identity = anIdentity; 
    }

    int Packet::getKind() const {
        return _kind; 
    }
            
    shared_ptr<Packet> Packet::getLink() { 
        return _link; 
    }
            
    void Packet::setLink(shared_ptr<Packet> aLink) { 
        _link = aLink;
    }

    string Packet::toString() const {
        return "Packet id: " + to_string(_identity) + " kind: " + to_string(_kind);
    }
}
